python/tidy/servo_tidy/tidy.py

```diff
@@ -510,6 +513,28 @@ def check_rust(file_name, lines):
                       + decl_found.format(crate_name))
             prev_crate[indent] = crate_name
 
+        # check alphabetical order of feature attributes in lib.rs files
+        if is_lib_rs_file:
+            match = re.search(r"#!\[feature\((.*)\)\]", line)
+
+            if match:
+                features = map(lambda w: w.strip(), match.group(1).split(','))
+                sorted_features = sorted(features)
+                if sorted_features != features:
+                    yield(idx + 1, decl_message.format("feature attribute")
+                          + decl_expected.format(tuple(sorted_features))
+                          + decl_found.format(tuple(features)))
+
+                if prev_feature_name > sorted_features[0]:
+                    yield(idx + 1, decl_message.format("feature attribute")
+                          + decl_expected.format(sorted_features[0] + " after " + prev_feature_name)
+                          + decl_found.format(prev_feature_name + " before " + sorted_features[0]))
```

Comment:
If we've found `foo before bar`, then how can we expect `bar after foo`? It's already after `foo`, right? Just swap `sorted_features[0]` and `prev_feature_name` in the first line, and we'll be good to go :)